front/app.py

```diff
@@ -2506,6 +2508,107 @@ def on_get(req, resp):
         resp.content_type = 'application/text'
         resp.text = (f"{ra}, {dec}")
 
+def checkFileAge():
+    if (os.path.exists('CometEls.txt') == False):
+        redownload = True
+    else:
+        creation_date = datetime.fromtimestamp(os.path.getctime('CometEls.txt'))
+        today = datetime.today()
+        delta = today - creation_date
+        if (delta.days > 7): 
+            redownload  = True
+        else:
+            redownload = False
+    return redownload
+
+
+def get_UTC_Time():
+    local_time = datetime.now(tzlocal.get_localzone())
+    utc_time = local_time - local_time.utcoffset()  # Manually adjust to UTC
+    return utc_time.year, utc_time.month, utc_time.day, utc_time.hour, utc_time.minute, utc_time.second
+
+def searchComet(name):
+    with load.open(mpc.COMET_URL, reload=checkFileAge()) as f:
+        comets = mpc.load_comets_dataframe(f)
+
+    # Keep only the most recent orbit for each comet,
+    # and index by designation for fast lookup.
+    comets = (comets.sort_values('reference')
+            .groupby('designation', as_index=False).last()
+            .set_index('designation', drop=False))
+
+    regex = re.compile(r'\b{}\b'.format(name), re.IGNORECASE)
+
+    row =  comets[comets['designation'].str.contains(regex)]
+    print(row['designation'].str)
+    if (len(row) == 0):
+        return 9999, 9999
+    ts = load.timescale()
+    eph = load('de440s.bsp')
+    sun, earth = eph['sun'], eph['earth']
+
+    comet = sun + mpc.comet_orbit(row.iloc[0], ts, GM_SUN)
+    Y,M,D,h,m,s = get_UTC_Time()
+    t = ts.utc(Y,M,D,h,m,s)
+    ra, dec, distance = earth.at(t).observe(comet).radec()
+
+    return str(ra).replace(" ",""), str(dec).replace('deg','d').replace("'","m").replace('"',"s").replace(" ",""), comet.target
+
+def searchMinorPlanet(name):
+    with load.open('https://www.minorplanetcenter.net/iau/Ephemerides/Bright/2024/Soft00Bright.txt') as f:
```

Comment:
We'll need to bump the URL in a few months, but that's a "future us" problem.

Reply:
Problem is they don't automatically release every month, sometimes it takes a while from what I've read.  If I could be more certain on releases I could just use datetime to put in the year.